### PR TITLE
[Port to 3.x] Form customization front end fixes/improvements

### LINKED ISSUE #15357: A label is cut off after applying form customization rule
### Summary
If you try to rename the label of any default field of the resource and use a comma in the text, it will fail. See images.
Screen Shot 2020-12-23 at 13 09 14
Screen Shot 2020-12-23 at 13 10 23

### Step to reproduce
Create any form of customization set and rule. In the rule, write a new name for any default field, like pagetitle, for example.
Use a comma in the name. After to go to the resource and check what label is shown.

### Observed behavior
Labels are cut off after the rule is applied.

### Expected behavior
Labels should be the same as it was set.

### Related issue(s)/PR(s)
N/A

### Environment
MODX 2.8.1

### PR DESCRIPTION
**Pull request by smg6511**
### What does it do?
Reworks the methods responsible for moving and changing field/tab visibility and labelling. Code optimizations were made and method documentation was added. Note this is a port of PR #15836, which will ultimately be withdrawn as it creates breaking changes not desirable for 2.x.

### Why is it needed?
Most of these methods were written to receive/process serialized/array data, but there appears to be no reason (any longer at least) to keep them that way—especially given that problems have arisen such as the one referenced by this PR. The changes allow for the use of commas in labels and provides simplification and clarification of the relevant code.

### How to test
Clear your caches and remember to run `grunt build` within the `_build/templates/default` directory to compile the changes into the main js file. Create one or two test TVs, then create a Form Customization profile with at least one rule. Within your rule(s), experiment with re-labelling and hiding various resource fields along with your test TVs. Also create a new tab in Regions tab and verify that assigning your TVs to that tab (or the standard regions) works as expected.

### Related issue(s)/PR(s)
Resolves #15357.

### CODE REVIEW

File: core/src/Revolution/modActionDom.php [hunk 1]
```diff
@@ -32,10 +32,10 @@ class modActionDom extends modAccessibleSimpleObject
      */
     public function apply($objId = '')
     {
+        /*
```
Comment (Mark-H) [inline core/src/Revolution/modActionDom.php]: Removing commented-out code would be preferred here.

Comment (smg6511) [reply core/src/Revolution/modActionDom.php]: Can't believe I forgot to delete that! Doing so now...

File: manager/assets/modext/core/modx.js [hunk 2]
```diff
@@ -478,33 +486,49 @@ Ext.extend(MODx,Ext.Component,{
...
+        if (tabPanel) {
+            const tabObj = tabPanel.getItem(regionId);
             if (tabObj) {
-                var z = tp.hideTabStripItem(tab);
-                MODx.hiddenTabs.push(tab);
-                var idx = this._getNextActiveTab(tp,tab);
-                tp.setActiveTab(idx);
+                // const z = tabPanel.hideTabStripItem(regionId),
+                //     idx = this._getNextActiveTab(tabPanel,regionId)
+                // ;
```
Comment (JoshuaLuckers) [inline manager/assets/modext/core/modx.js]:
```suggestion
```

Removing code is preferred

File: manager/assets/modext/core/modx.js [hunk 3]
```diff
@@ -478,33 +486,49 @@ Ext.extend(MODx,Ext.Component,{
...
+                tabPanel.hideTabStripItem(regionId);
+                MODx.hiddenTabs.push(regionId);
+                // tabPanel.setActiveTab(idx);
```
Comment (JoshuaLuckers) [inline manager/assets/modext/core/modx.js]:
```suggestion
```

Removing code is preferred

**Review by Ibochkarev — COMMENTED** [review]
In my test, the customization of forms does not work at all, neither moving through areas, nor changing TV, nor setting the default value.

What am I doing wrong?

https://user-images.githubusercontent.com/2138260/142778757-1960ed35-bc53-42e0-aa93-ff834c4b5351.mp4

**Comment by smg6511** [comment]
@Ibochkarev - The likely solution is to clear your browser cache. I had the same thing happen when moving over to this branch to check out what the problem could be: My browser was using a cached version of the js files.

File: manager/assets/modext/core/modx.js (hunk 2 above)
Comment (smg6511) [reply manager/assets/modext/core/modx.js]: Fixed.

File: manager/assets/modext/core/modx.js (hunk 3 above)
Comment (smg6511) [reply manager/assets/modext/core/modx.js]: Fixed.

**Review by theboxer — APPROVED** [review]

**Pull request merged**